lib/index.js

```diff
@@ -65,7 +65,7 @@ exports.scan = function (obj, options) {
         next = [];
 
         for (const node of nodes) {
-            if (node.hasOwnProperty('__proto__')) {
+            if (Object.prototype.hasOwnProperty.call(node, '__proto__')) {
```

Comment:
I would recommend doing `const hasOwnProperty = Object.prototype.hasOwnProperty;` at the top of the file, and then reuse that. It's slightly more secure.